src/Microsoft.ML.Data/StaticPipe/TrainerEstimatorReconciler.cs

```diff
@@ -0,0 +1,335 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+namespace Microsoft.ML.Data.StaticPipe.Runtime
+{
+    /// <summary>
+    /// General purpose reconciler for a typical case with trainers, where they accept some generally
```

Comment:
>reconciler [](start = 24, length = 10)

I am still fuzzy on the purpose of reconcilers. is it a bridge from estimators to trainers?